src/ast/CREATE_DECK.ts

```diff
@@ -0,0 +1,56 @@
+import NODE from '../lib/NODE';
+import DECK from './DECK';
+import TAGS from './TAGS';
+import STYLE from './STYLE';
+
+const invalidNameTokens = ["with Tags: ", "using Style: ", "("];
+
+export default class CREATE_DECK extends NODE {
+    // style: STYLE;
+    tags: TAGS | null = null;
+    style: STYLE | null = null;
+    deck: DECK | null = null;
+    name: string = "";
+
+    checkForAndParseTags(): boolean {
+        if (this.tokenizer.checkNext("with Tags: ")) {
```

Comment:
current checkNext() in tokenizer takes no parameters and is used to see if there's any more tokens, maybe use checkToken(regex)

Reply:
oh you're right. It should have been checkToken()